courses/fundamentals_of_ada/adv_250_containers.rst

```diff
@@ -35,69 +35,255 @@ Introduction
 Container Library
 -------------------
 
-* Packages (including generics) for different types of data containers
+* :ada:`Ada.Containers` parent package
+* Packages (including generics)
 
-   - Vectors
-   - Doubly-linked lists
-   - Maps
-   - Sets
-   - Trees
-   - Queues
-   - Indefinite containers (holder/wrapper)
+    - Different types of data containers
+    - Hold an :ada:`Element` type
+    - Container types are :ada:`tagged`
 
-* Sorting mechanisms
+* Types defined as a product of both
+
+    - A data structure
+    - An implementation
+    - Define some added operations
+
+* Containers share sets of operations
+
+    - Seen later
+
+=================
+Container Types
+=================
+
+-----------------------
+Data Structures (1/2)
+-----------------------
+
+* Vector
+
+   - Essentially an array
+   - :dfn:`Capacity` and size can differ
```

Comment:
You should either explain the difference or, if explained later, remove it from here

Reply:
Indeed, also size was improper, the most common name is length